sql/sql_alter.cc

```diff
@@ -84,6 +84,24 @@ bool Alter_info::set_requested_lock(const LEX_STRING *str)
   return false;
 }
 
+/**
+  Checks if there are any columns with COLUMN_FORMAT COMRPESSED
+  attribute among field definitions in create_list.
+
+  @retval false there are no compressed columns
+  @retval true there is at least one compressed column
+*/
+bool Alter_info::has_compressed_columns() const
+{
+  List_iterator<Create_field> it(
+    const_cast<List<Create_field>&>(create_list));
+  Create_field* sql_field = it++;
```

Comment:
1. In this particular case the list cannot be empty (empty field list is not allowed).
2.  `it++` for a iterator created from an empty list is a valid construction and will return NULL (base list has a global termination node `list_node end_of_list` for such cases)